`ipfs swarm peering ls` crashes if daemon isn't running

### Checklist

- [X] This is a bug report, not a question. Ask questions on [discuss.ipfs.io](https://discuss.ipfs.io).
- [X] I have searched on the [issue tracker](https://github.com/ipfs/go-ipfs/issues?q=is%3Aissue) for my bug.
- [X] I am running the latest [go-ipfs version](https://dist.ipfs.io/#go-ipfs) or have an issue updating.

### Installation method

third-party binary

### Version

```Text
go-ipfs version: 0.11.0
Repo version: 11
System version: amd64/darwin
Golang version: go1.16.12
```


### Config

```json
{
  "API": {
    "HTTPHeaders": {}
  },
  "Addresses": {
    "API": "/ip4/127.0.0.1/tcp/5001",
    "Announce": [],
    "AppendAnnounce": [],
    "Gateway": "/ip4/127.0.0.1/tcp/8080",
    "NoAnnounce": [],
    "Swarm": [
      "/ip4/0.0.0.0/tcp/4001",
      "/ip6/::/tcp/4001",
      "/ip4/0.0.0.0/udp/4001/quic",
      "/ip6/::/udp/4001/quic"
    ]
  },
  "AutoNAT": {},
  "Bootstrap": [
    "<REDACTED>",
  ],
  "DNS": {
    "Resolvers": {}
  },
  "Datastore": {
    "BloomFilterSize": 0,
    "GCPeriod": "1h",
    "HashOnRead": false,
    "Spec": {
      "mounts": [
        {
          "child": {
            "path": "blocks",
            "shardFunc": "/repo/flatfs/shard/v1/next-to-last/2",
            "sync": true,
            "type": "flatfs"
          },
          "mountpoint": "/blocks",
          "prefix": "flatfs.datastore",
          "type": "measure"
        },
        {
          "child": {
            "compression": "none",
            "path": "datastore",
            "type": "levelds"
          },
          "mountpoint": "/",
          "prefix": "leveldb.datastore",
          "type": "measure"
        }
      ],
      "type": "mount"
    },
    "StorageGCWatermark": 90,
    "StorageMax": "10GB"
  },
  "Discovery": {
    "MDNS": {
      "Enabled": true,
      "Interval": 10
    }
  },
  "Experimental": {
    "AcceleratedDHTClient": false,
    "FilestoreEnabled": false,
    "GraphsyncEnabled": false,
    "Libp2pStreamMounting": false,
    "P2pHttpProxy": false,
    "StrategicProviding": false,
    "UrlstoreEnabled": false
  },
  "Gateway": {
    "APICommands": [],
    "HTTPHeaders": {
      "Access-Control-Allow-Headers": [
        "X-Requested-With",
        "Range",
        "User-Agent"
      ],
      "Access-Control-Allow-Methods": [
        "GET"
      ],
      "Access-Control-Allow-Origin": [
        "*"
      ]
    },
    "NoDNSLink": false,
    "NoFetch": false,
    "PathPrefixes": [],
    "PublicGateways": null,
    "RootRedirect": "",
    "Writable": false
  },
  "Identity": {
    "PeerID": "<REDACTED>"
  },
  "Internal": {},
  "Ipns": {
    "RecordLifetime": "",
    "RepublishPeriod": "",
    "ResolveCacheSize": 128
  },
  "Migration": {
    "DownloadSources": [],
    "Keep": ""
  },
  "Mounts": {
    "FuseAllowOther": false,
    "IPFS": "/ipfs",
    "IPNS": "/ipns"
  },
  "Peering": {
    "Peers": null
  },
  "Pinning": {
    "RemoteServices": {}
  },
  "Plugins": {
    "Plugins": null
  },
  "Provider": {
    "Strategy": ""
  },
  "Pubsub": {
    "DisableSigning": false,
    "Router": ""
  },
  "Reprovider": {
    "Interval": "12h",
    "Strategy": "all"
  },
  "Routing": {
    "Type": "dht"
  },
  "Swarm": {
    "AddrFilters": null,
    "ConnMgr": {
      "GracePeriod": "20s",
      "HighWater": 900,
      "LowWater": 600,
      "Type": "basic"
    },
    "DisableBandwidthMetrics": false,
    "DisableNatPortMap": false,
    "RelayClient": {},
    "RelayService": {},
    "Transports": {
      "Multiplexers": {},
      "Network": {},
      "Security": {}
    }
  }
}
```


### Description

$ ipfs swarm peering ls


panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x1d1d966]

goroutine 1 [running]:
github.com/ipfs/go-ipfs/peering.(*PeeringService).ListPeers(0x0, 0x0, 0x0, 0x0)
        github.com/ipfs/go-ipfs@v0.11.0/peering/peering.go:247 +0x66
github.com/ipfs/go-ipfs/core/commands.glob..func167(0xc000266000, 0x2a1f870, 0xc000339260, 0x240f740, 0xc000339200, 0x1, 0xc000339260)
        github.com/ipfs/go-ipfs@v0.11.0/core/commands/swarm.go:149 +0x77
github.com/ipfs/go-ipfs-cmds.(*executor).Execute(0xc00000e730, 0xc000266000, 0x2a1f870, 0xc000339260, 0x240f740, 0xc000339200, 0xc000339260, 0x0)
        github.com/ipfs/go-ipfs-cmds@v0.6.0/executor.go:77 +0x16b
github.com/ipfs/go-ipfs-cmds/cli.Run(0x2a1f3d8, 0xc00020dc00, 0x3766920, 0xc0000d4040, 0x4, 0x4, 0xc0000d0000, 0xc0000d0008, 0xc0000d0010, 0x2857768, ...)
        github.com/ipfs/go-ipfs-cmds@v0.6.0/cli/run.go:137 +0xab8
main.mainRet(0x0)
        github.com/ipfs/go-ipfs@v0.11.0/cmd/ipfs/main.go:168 +0x45e
main.main()
        github.com/ipfs/go-ipfs@v0.11.0/cmd/ipfs/main.go:71 +0x25

--- a/core/commands/swarm.go
+++ b/core/commands/swarm.go
@@ -850,7 +850,7 @@ Filters default to those specified under the "Swarm.AddrFilters" config key.
 			return err
 		}
 
-		if n.PeerHost == nil {
+		if !n.IsOnline {
 			return ErrNotOnline
 		}
 
@@ -886,7 +886,7 @@ var swarmFiltersAddCmd = &cmds.Command{
 			return err
 		}
 
-		if n.PeerHost == nil {
+		if !n.IsOnline {
 			return ErrNotOnline
 		}
 
@@ -942,7 +942,7 @@ var swarmFiltersRmCmd = &cmds.Command{
 			return err
 		}
 
-		if n.PeerHost == nil {
+		if !n.IsOnline {
 			return ErrNotOnline
 		}
 

--- a/core/commands/swarm.go
+++ b/core/commands/swarm.go
@@ -216,7 +216,7 @@ var swarmPeeringRmCmd = &cmds.Command{
 	Type: peeringResult{},
 	Encoders: cmds.EncoderMap{
 		cmds.Text: cmds.MakeTypedEncoder(func(req *cmds.Request, w io.Writer, pr *peeringResult) error {
-			fmt.Fprintf(w, "add %s %s\n", pr.ID.String(), pr.Status)
+			fmt.Fprintf(w, "remove %s %s\n", pr.ID.String(), pr.Status)
 			return nil
 		}),
 	},

--- a/core/commands/swarm.go
+++ b/core/commands/swarm.go
@@ -122,7 +122,7 @@ var swarmPeeringAddCmd = &cmds.Command{
 		if err != nil {
 			return err
 		}
-		if node.PeerHost == nil {
+		if !node.IsOnline {
 			return ErrNotOnline
 		}
 
@@ -156,7 +156,7 @@ var swarmPeeringLsCmd = &cmds.Command{
 		if err != nil {
 			return err
 		}
-		if node.PeerHost == nil {
+		if !node.IsOnline {
 			return ErrNotOnline
 		}
 
@@ -196,7 +196,7 @@ var swarmPeeringRmCmd = &cmds.Command{
 		if err != nil {
 			return err
 		}
-		if node.PeerHost == nil {
+		if !node.IsOnline {
 			return ErrNotOnline
 		}
 

--- a/core/commands/swarm.go
+++ b/core/commands/swarm.go
@@ -122,6 +122,9 @@ var swarmPeeringAddCmd = &cmds.Command{
 		if err != nil {
 			return err
 		}
+		if node.PeerHost == nil {
+			return ErrNotOnline
+		}
 
 		for _, addrinfo := range addInfos {
 			node.Peering.AddPeer(addrinfo)
@@ -153,6 +156,10 @@ var swarmPeeringLsCmd = &cmds.Command{
 		if err != nil {
 			return err
 		}
+		if node.PeerHost == nil {
+			return ErrNotOnline
+		}
+
 		peers := node.Peering.ListPeers()
 		return cmds.EmitOnce(res, addrInfos{Peers: peers})
 	},
@@ -188,6 +195,9 @@ var swarmPeeringRmCmd = &cmds.Command{
 		node, err := cmdenv.GetNode(env)
 		if err != nil {
 			return err
+		}
+		if node.PeerHost == nil {
+			return ErrNotOnline
 		}
 
 		for _, arg := range req.Arguments {

